src/cast_core/src/CrudActor.cpp

```diff
@@ -239,16 +272,58 @@ auto createGenerator(YAML::Node source,
 
 }  // namespace
 
+struct CreateIndexOperation : public BaseOperation {
+    mongocxx::collection _collection;
```

Comment:
Can we push these guys up to `BaseOperation`, maybe as a variadic template to save more typing? I think every CRUD operation needs `collection`, `operation (for metrics)` and `onSession`.

Reply:
Also, technically... `CreateIndexOperation` isn't CRUD. Feel free to leave it in here but add a comment. Alternatively, to force an error, you can insert two documents with the same `_id` field since the `_id` index is automatically created and is unique.

Reply:
> Can we push these guys up to BaseOperation, maybe as a variadic template to save more typing?

Somehow I missed this comment before I added the remainder of the operations. I'm going to punt on this for now but will add a comment that it could likely be refactored in the future.

> Also, technically... CreateIndexOperation isn't CRUD. Feel free to leave it in here but add a comment....

Oh..ya..nice..I wish I'd thought of the same `_id` thing before implementing createindex. But it's here now so I'm gonna leave it. I added a comment per your suggestion.